File: 1603-design-parking-system/1603-design-parking-system.c



typedef struct {
    int small;
    int medium;
    int big;
    
} ParkingSystem;


ParkingSystem* parkingSystemCreate(int big, int medium, int small) {
    
    ParkingSystem *P = (ParkingSystem*)malloc(sizeof(ParkingSystem));
    P ->small = small;
    P ->medium = medium;
    P ->big = big;
    return P;
}

bool parkingSystemAddCar(ParkingSystem* obj, int carType) {
     
    if(carType == 3 && obj -> small > 0){
        
        obj -> small --;
        return true;
    }
    else if (carType == 2 && obj -> medium > 0){
        obj -> medium --;
        return true;
    }
    else if (carType == 1 && obj -> big > 0){
        obj -> big --;
        return true;
    }
    return false;
    
    
    
}

void parkingSystemFree(ParkingSystem* obj) {

    free(obj);
    
}

/**
 * Your ParkingSystem struct will be instantiated and called as such:
 * ParkingSystem* obj = parkingSystemCreate(big, medium, small);
 * bool param_1 = parkingSystemAddCar(obj, carType);
 
 * parkingSystemFree(obj);
*/

File: 1603-design-parking-system/1603-design-parking-system.java
class ParkingSystem {
    int big,medium,small;
    public ParkingSystem(int big, int medium, int small) {
        this.big=big;
        this.medium=medium;
        this.small=small;
    }

    public boolean addCar(int carType) {
        if(big>=1 && carType==1){
            big--;
            return true;
        }
        if(medium>=1 && carType==2){
            medium--;
            return true;
        }
        if(small>=1 && carType==3){
            small--;
            return true;
        }
        return false;
    }
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem obj = new ParkingSystem(big, medium, small);
 * boolean param_1 = obj.addCar(carType);
 */